tests/test_common.py

```diff
@@ -175,3 +181,12 @@ def test_render_raw_permission():
         '(indic0 == "0" or indic1 == 1) and ' 'indic2 == "yo_2" and indic_list == [\'0\', 1, \'2\']'
     )
     assert apply_query_parameters(query, params) == expected
+
+
+def test_bad_variable_in_query():
+    """It should thrown a NonValidEndpointVariable exception if bad variable in endpoint"""
+    query = {'url': '/stuff/%(thing)s/foo'}
+    params = {}
+    with pytest.raises(NonValidEndpointVariable) as err:
```

Comment:
I recently discovered that pytest.raises can match on the content on the expression:
https://docs.pytest.org/en/stable/assert.html#assertions-about-expected-exceptions

It's a nice and concise way to do the same thing, so it's not mandatory to merge this PR